pandas/core/indexes/multi.py

```diff
@@ -3433,3 +3433,26 @@ def maybe_droplevels(index, key):
             pass
 
     return index
+
+
+def _coerce_indexer_frozen(array_like, categories, copy: bool = False) -> np.ndarray:
```

Comment:
i would move this to pandas.core.dtypes.cast

Reply:
its only ever used here, and doesnt really have anything to do with dtypes